File: javascript/while-loop.js
// npm install prompt-sync
const prompt = require('prompt-sync')();

var running = true;

 while (running) {
 	const solution = Math.floor(Math.random() * 100) + 1;
 	var correct = false;

	console.log("You have 10 tries to guess my number between 1 and 100.\n");
	
	for (let i = 0; i < 10; i++) {
		let guess = prompt(`[${i}/10] What's your guess? `);
		if (guess < solution) {
			console.log("My number is bigger.");
		} else if (guess > solution) {
			console.log("My number is smaller.");
		} else {
			correct = true
			console.log("That's it. You guessed it.");
			break;
		}
	}

	if (!correct) {
		console.log("Ohh you didn't guessed it. :(")
	}

	running = prompt("Do you want to play another round? ") == "yes"
 }

File: go/while-loop.go
package main

import (
	"fmt"
	"math/rand"
	"time"
)

func main() {
	running := true

	for running {
		generator := rand.New(rand.NewSource(time.Now().UnixNano()))
		solution := generator.Intn(100)+1
		correct := false

		fmt.Println("You have 10 tries to guess my number between 1 and 100.\n")

		for i := 0; i < 10; i++ {
			fmt.Println("What's your guess? ")

			var guess int
			fmt.Scanln(&guess)
			
			if guess < solution {
				fmt.Println("My number is bigger.")
			} else if guess > solution {
				fmt.Println("My number is smaller.")
			} else {
				correct = true
				fmt.Println("That's it. You guessed it.")
				break
			}
		}

		if !correct {
			fmt.Println("Ohh you didn't guess it. :(")
		}

		fmt.Println("Do you want to play another round?")
		var continueGame string
		fmt.Scanln(&continueGame)

		running = continueGame == "yes"
	}
}